daemon/src/model/cfd.rs

```diff
@@ -2373,6 +2390,219 @@ mod tests {
+                FundingFee::dummy(),
+            ))
+            .unwrap_err();
+        assert_eq!(no_rollover_reason, NoRolloverReason::Closed);
+    }
+
+    /// Cover scenario where trigger a collab settlement during ongoing rollover
+    ///
+    /// In this scenario the collab settlement is still ongoing when the rollover finishes.
+    /// If we trigger a collaborative settlement during rollover, the settlement will have priority
+    /// over the rollover. Upon finishing the rollover we fail because the cfd was already
+    /// settled.
+    #[test]
+    fn given_ongoing_collab_settlement_then_cannot_finish_rollover() {
+        let quantity = Usd::new(dec!(10));
+        let leverage = Leverage::new(2).unwrap();
+        let opening_price = Price::new(dec!(10000)).unwrap();
+
+        let cfd = Cfd::taker_long()
+            .with_quantity(quantity)
+            .with_opening_price(opening_price)
+            .with_leverage(leverage)
+            .dummy_open(dummy_event_id());
+
+        let _ = assert!(cfd.start_rollover().is_ok(), "Could not start rollover");
+
+        let cfd = cfd.dummy_start_collab_settlement();
+
+        let no_rollover_reason = cfd
+            .roll_over(RolloverCompleted::succeeded(
+                OrderId::default(),
+                Dlc::dummy(None),
+                FundingFee::dummy(),
+            ))
+            .unwrap_err();
+        assert_eq!(
+            no_rollover_reason,
+            NoRolloverReason::InCollaborativeSettlement
+        );
+    }
+
+    #[test]
+    fn given_ongoing_collab_settlement_then_cannot_start_rollover() {
+        let quantity = Usd::new(dec!(10));
+        let leverage = Leverage::new(2).unwrap();
+        let opening_price = Price::new(dec!(10000)).unwrap();
+
+        let cfd = Cfd::taker_long()
+            .with_quantity(quantity)
+            .with_opening_price(opening_price)
+            .with_leverage(leverage)
+            .dummy_open(dummy_event_id());
+
+        let _ = assert!(cfd.start_rollover().is_ok(), "Could not start rollover");
+
+        let cfd = cfd.dummy_start_collab_settlement();
+
+        let err = cfd.start_rollover().unwrap_err();
+        let no_rollover_reason = err.downcast::<NoRolloverReason>().unwrap();
+        assert_eq!(
+            no_rollover_reason,
+            NoRolloverReason::InCollaborativeSettlement
+        );
+
+        let cfd = Cfd::maker_short()
+            .with_quantity(quantity)
+            .with_opening_price(opening_price)
+            .with_leverage(leverage)
+            .dummy_open(dummy_event_id());
+
+        let _ = assert!(cfd.start_rollover().is_ok(), "Could not start rollover");
+
+        let cfd = cfd.dummy_start_collab_settlement();
+
+        let err = cfd.start_rollover().unwrap_err();
+        let no_rollover_reason = err.downcast::<NoRolloverReason>().unwrap();
+        assert_eq!(
+            no_rollover_reason,
+            NoRolloverReason::InCollaborativeSettlement
```

Comment:
This is a completely separate test, isn't it? (the maker side of things)

Reply:
Yes, but the same test context. I could split it up in `_taker` and `_maker` test but that goes towards the same discussion of being able to group.